mobile/src/calendar/calendar-legend.tsx

```diff
@@ -0,0 +1,66 @@
+import React, { Component } from 'react';
+import { View, StyleSheet, PixelRatio } from 'react-native';
+import { connect } from 'react-redux';
+import { IntervalsModel, WeekModel } from '../reducers/calendar/calendar.model';
+import { AppState } from '../reducers/app.reducer';
+import { CalendarEventsType } from '../reducers/calendar/calendar-events.model';
+import { CalendarIntervalColor, legendStyles as styles } from './styles';
+import { StyledText } from '../override/styled-text';
+
+interface CalendarLegendProps {
+    intervals: IntervalsModel;
+    weeks: WeekModel[];
+}
+
+const mapStateToProps = (state: AppState): CalendarLegendProps => ({
+    intervals: state.calendar.calendarEvents.intervals,
+    weeks: state.calendar.calendarEvents.weeks
+});
+
+class CalendarLegendImpl extends Component<CalendarLegendProps> {
+    public render() {
+        const existingIntervals = this.getVisibleCalendarEvents();
+        const legend: JSX.Element[] = [];
+
+        existingIntervals.forEach((value, type) => {
+            const style = StyleSheet.flatten([styles.marker, { backgroundColor: CalendarIntervalColor.getColor(type) }]);
+            legend.push((
+                <View key={type} style={styles.itemContainer}>
+                    <View style={style}></View>
+                    <StyledText style={styles.label}>{type}</StyledText>
+                </View>
+            ));
+        });
+
+        return (
+            <View style={styles.container}>
+                <View>
+                    {legend}
+                </View>
+            </View>
+        );
+    }
+
+    private getVisibleCalendarEvents() {
+        const { intervals, weeks } = this.props;
+        const calendarEventsMap: Map<CalendarEventsType, number> = new Map();
+
+        if (!weeks || !intervals) {
+            return calendarEventsMap;
+        }
+
+        weeks.forEach(week => {
```

Comment:
I would put this logic to ```IntervalsModel```, for example to ```eventTypesByMonth()```

Reply:
Isn't it view specific logic that might not be required anywhere else?

Reply:
I think it looks like a part of IntervalsModel api. And doesn't depend on whether you are going to reuse it or not, having it in component is difficult to test it.

Reply:
@santee